cmd/kubeadm/app/apis/kubeadm/v1beta1/doc.go

```diff
@@ -19,14 +19,24 @@ limitations under the License.
 // +k8s:deepcopy-gen=package
 // +k8s:conversion-gen=k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm
 
-// Package v1beta1 defines the v1beta1 version of the kubeadm config file format, that is a big step
-// forward the objective of graduate kubeadm config to beta.
-//
-// //TODO add notes about big changes introduced by this release
+// Package v1beta1 defines the v1beta1 version of the kubeadm config file format.
+// This version graduates the kubeadm config to BETA and is a big step towards GA.
+//
+// A list of changes since v1alpha3:
+// - "apiServerEndpoint" in InitConfiguration was renamed to "localAPIServerEndpoint" for better clarity of what the field
+// represents.
+// - Common fields in ClusterConfiguration such as "*extraArgs" and "*extraVolumes" for control plane components are now moved
+// under component structs - i.e. "apiServer", "controllerManager", "scheduler".
+// - "auditPolicy" was removed from ClusterConfiguration. Please use "extraArgs" in "apiServer" to configure this feature instead.
+// - "unifiedControlPlaneImage" in ClusterConfiguration was changed to a boolean field called "useHyperKubeImage".
+// - ClusterConfiguration now has a "dns" field which can be used to select and configure the cluster DNS addon.
+// - Both "localEtcd" and "dns" configurations now support custom image repositories.
+// - the "controlPlane*" related fields in JoinConfiguration were refactored into a sub structure.
+// - "clusterName" was removed from JoinConfiguration and the name is now fetched from the existing cluster.
```

Comment:
We should probably mention `featureGates` too here.

Reply:
i did not mention them as we already have release notes on them and the "featureGates" element has not changed as a container for them.
but we can definitely do that, i guess.